StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/FileHeaderCodeFixProvider.cs

```diff
@@ -106,8 +140,32 @@ private static SyntaxNode ReplaceHeader(Document document, SyntaxNode root, Styl
                 }
             }
 
+            // Remove copyright lines in reverse order
+            removalList.Reverse();
+            foreach (var triviaLine in removalList)
+            {
+                trivia = trivia.RemoveAt(triviaLine);
+            }
```

Comment:
This can be implemented more efficiently using an for loop with a decreasing indexer.